actionpack/lib/action_dispatch/middleware/debug_exceptions.rb

```diff
@@ -181,5 +184,9 @@ def routes_inspector(exception)
         ActionDispatch::Routing::RoutesInspector.new(@routes_app.routes.routes)
       end
     end
+
+    def api_request?
+      @response_format == :api
```

Comment:
I'd make this be `@response_format == :api && !content_type.html?` and then remove the conditional inside  `render_for_api_request` method.